### PR TITLE
Automate filling in the main exception list

### LINKED ISSUE #2632: All config/prod/prow/* config needs to be not hardcoded to master.
Move to not hardcode the HEAD branch of any repo. This is an effort to support inclusive naming and eliminate the usage of a `master` branch.

See https://github.com/knative/community/issues/293﻿

### PR DESCRIPTION
**Pull request by markusthoemmes**
Fixes #2632

**What this PR does, why we need it**:

A followup to https://github.com/knative/test-infra/pull/2683 where the list of main the default branches will be filled automatically for the `knative` and `knative-sandbox` org, so we "just" need to rerun the generator once flipping the respective switches.

Ideally, we'd send a request per repo and pick whatever default branch the respective repo has, but that makes us run into rate-limiting immediately and this seems to be a good 80% solution while we're in migration.

### CODE REVIEW

File: tools/config-generator/main.go [hunk 1]
```diff
@@ -760,6 +766,23 @@ func main() {
 		}
 	}
 
+	// Fill in the main exception list. We fetch this once for all of our orgs to avoid
+	// rate-limiting issues with Github.
+	client := github.NewClient(nil)
+	for org := range pathAliasOrgs {
+		repos, _, err := client.Repositories.List(context.Background(), org, &github.RepositoryListOptions{
```
Comment (chaodaiG) [inline tools/config-generator/main.go:773]: https://github.com/knative/test-infra/blob/e20e6a9df5009eb74eed35b453fcdb25cdb65f0a/pkg/ghutil/repo.go#L28
Can we use this instead?

Comment (markusthoemmes) [reply tools/config-generator/main.go:773]: Well, nope, cause it actually needs the entire response :thinking:

**Comment by chaodaiG** [comment]
There is a merge conflict, will approve once it's fixed

**Review by chaodaiG — APPROVED** [review]
/lgtm

**Pull request merged**